viewer/packages/data-providers/src/image-360-data-providers/descriptor-providers/events/Cdf360BatchEventCollectionLoader.test.ts

```diff
@@ -0,0 +1,200 @@
+      'Events API failed'
+    );
+  });
+
+  test('should filter out incomplete file sets', async () => {
+    const events = [createMockEvent('site1', 'station1', 1)];
+    // Only 4 files instead of 6 - incomplete set
+    const incompleteFiles = createMockFiles('site1', 'station1', 100).slice(0, 4);
+
+    const clientMock = createMockClient(events, incompleteFiles);
+    const batchLoader = new Cdf360BatchEventCollectionLoader(clientMock);
+    const result = await batchLoader.getCollectionDescriptors({ site_id: 'site1' }, false);
+
+    // Should return empty because file set is incomplete (needs 6 faces, only has 4)
+    expect(result.length).toBe(0);
+  });
+
+  test('should deduplicate requests for the same site within batch window', async () => {
+    const events = [createMockEvent('site1', 'station1', 1)];
+    const files = createMockFiles('site1', 'station1', 100);
+
+    let batchExecutionCount = 0;
+    const clientMock = createMockClient(events, files, () => batchExecutionCount++);
+    const batchLoader = new Cdf360BatchEventCollectionLoader(clientMock);
+
+    // Make 3 requests for the same site - should be deduplicated in the batch
+    await Promise.all([
+      batchLoader.getCollectionDescriptors({ site_id: 'site1' }, false),
+      batchLoader.getCollectionDescriptors({ site_id: 'site1' }, false),
+      batchLoader.getCollectionDescriptors({ site_id: 'site1' }, false)
+    ]);
+
+    // Should execute only once (10 partitions), not 3 times (30 calls)
+    // This validates that duplicate site_ids within a batch are deduplicated
+    expect(batchExecutionCount).toBe(10);
+  });
+
+  function createMockClient(events: CogniteEvent[], files: FileInfo[], onEventsList?: () => void) {
+    const eventsMock = new Mock<ReturnType<CogniteClient['events']['list']>>()
+      .setup(instance => instance.autoPagingToArray)
+      .returns(async () => events)
+      .object();
+
+    const filesMock = new Mock<ReturnType<CogniteClient['files']['list']>>()
+      .setup(instance => instance.autoPagingEach)
+      .returns(async (callback: (file: FileInfo) => void) => {
+        files.forEach(callback);
+      })
+      .object();
+
+    const eventsApiMock = new Mock<CogniteClient['events']>()
+      .setup(instance => instance.list)
+      .returns(() => {
+        onEventsList?.();
+        return eventsMock;
+      })
+      .object();
+
+    const filesApiMock = new Mock<CogniteClient['files']>()
+      .setup(instance => instance.list)
+      .returns(() => filesMock)
+      .object();
+
+    return new Mock<CogniteClient>()
+      .setup(instance => instance.events)
+      .returns(eventsApiMock)
+      .setup(instance => instance.files)
+      .returns(filesApiMock)
+      .object();
+  }
+
+  function createMockClientWithError(errorMessage: string) {
+    const eventsMock = new Mock<ReturnType<CogniteClient['events']['list']>>()
+      .setup(instance => instance.autoPagingToArray)
+      .returns(async () => {
```

Comment:
There is some duplication between this function and the one above.. Can we do something to deduplicate them a bit?